### DISCUSSION TITLE
Navigation actions new design

### DISCUSSION THREAD

**Discussion by devemux86** [devemux86/cruiser#57] [Q&A] [answered]
We can redesign the navigation **actions** screen with large icon buttons for better usability.
(appears when **press** the top left navigation panel)

Please let me know if you think the button order should be different.

---

|Skip next waypoint|Skip waypoints|
|:-:|:-:|
|**Avoid roadblock**|**Route overview**|

<img src="https://user-images.githubusercontent.com/3484020/222767346-f304b504-3ce4-412d-b7ab-bd5f045e20a1.jpg" width="50%"/>
<img src="https://user-images.githubusercontent.com/3484020/222767360-5bd83c7c-1fd8-47dd-acf3-161f78c43355.jpg"/>

**Comment by Leeb64** [comment]
Generell könnten die Touchfelder in den Navigationsaktionen etwas besser gestaltet werden, um sie besser erkennbar zu machen. Vielleicht mit einem fetten Rahmen oder anderen Farben...
Vielleicht etwas größer, um Fehlbedienungen zu vermeiden - am Motorrad mit Handschuhen.

**Reply by devemux86** [reply]
Indeed, this is the purpose of the new design.
Large buttons with colored icons and make them easier to press with gloves.

---

In der Tat ist dies der Zweck des neuen Designs.
Große Tasten mit farbigen Symbolen und machen sie leichter mit Handschuhen zu drücken.

**Comment by WolfgangZ42** [comment]
The last one ("Route Overview") is redundant to the short tap on the compass sign. 
What about to offer a screenshot instead? Just after closing the sub-panel and with the current timestamp into the same directory as the GPS recordings are stored.

**Reply by devemux86** [reply]
We could free the "tap on compass", as the compass may auto hide,
so it can be more difficult for a quick route overview?

And instead use the tap to show the navigation orientation options: rotation 2D/3D or not.
(now with "long press on compass")

**Reply by devemux86** [reply]
> What about to offer a screenshot instead?

Nice idea, Calimoto has an extra button.

A problem may be new Androids require file operations to ask for user permission, unless writing in app internal storage.
e.g. import / export functions use Android file browser for this purpose

**Reply by WolfgangZ42** [reply]
Not only Calimoto has an extra button - so I have it via an overlay app for screenshots. 😅 

Okay - considering the problem with "user permission" and the possibility of auto hide for the compass sign, I plead for the selection of 2D/3D/no.

**Comment by DolomitesTom** [comment]
Yes, large buttons, already long awaited... but better without extra permissions. If the size can be defined via setting or size slider, then a cancel or back button is also needed in case it fills the full screen. And some space for the already described WOW button was also not bad 😎

**Comment by MotoUKRider** [comment]
This is a good direction for the UI - for Cruiser to stand a chance against other (motorcycle) apps, it should be easy to use "in the saddle". A few popular big buttons would certainly help.  It's still a bit frustrating to use in landscape mode (eg: waypoints list, especially the Via points with mile/time info), but that's what rotating RAM-mounts were invented for.

**Comment by devemux86** [comment]
See edited 1st post for preview images (portrait / landscape): https://github.com/devemux86/cruiser/discussions/57#discussion-4894028

(texts are kept to a minimum to avoid overflow with large fonts)

**Comment by devemux86** [comment]
New preview images with colored buttons (portrait / landscape) in https://github.com/devemux86/cruiser/discussions/57#discussion-4894028.

We could increase the actions if necessary in the future.

**Comment by airald** [comment]
This will be a very useful feature. 
From many time ago, apps are using simpler color schemes. Usually, one color for all buttons. Visually is good in my oppinion, but if each button have a different color, is more easy to tap de desired option.

**Reply by devemux86** [reply]
> Visually is good in my oppinion, but if each button have a different color, is more easy to tap de desired option.

Indeed and we should not forget that there is also the _dark app theme_ with slightly different colors.
Let's wait for the next release and you can all test it and suggest improvements in colors or sizes.

Different button colors could also be implemented. :slightly_smiling_face:

**Comment by devemux86** [comment, marked as answer]
The new navigation actions screen is available in [Cruiser 3.0.11](https://play.google.com/store/apps/details?id=gr.talent.cruiser).
(tap navigation panel)

Please let me know how you see it in both the light and dark app themes.

There is already a suggestion to put different colors on the buttons.

**Reply by DolomitesTom** [reply]
look very nice in both color schemes

**Comment by Leeb64** [comment]
At Skip ? :
When skipping waypoints, the numbering is incorrect.
Skip WP1->Nothing Happens
Skip WP2-> WP1 is skipped etc.

**Reply by devemux86** [reply]
@Leeb64 please see https://github.com/devemux86/cruiser/discussions/17#discussioncomment-5215938 for the answer in the feature's topic.

**Comment by DolomitesTom** [comment]
it is working here without problems (brouter offline)
